olclient/openlibrary.py

```diff
@@ -732,6 +819,27 @@ def _generate_url_from_olid(self, olid):
         return "%s/%s/%s.json" % (self.base_url, ol_paths[kind], olid)
 
     @staticmethod
+    def get_text_value(text):
+        """Returns the text value from a property that can either be a properly
+        formed /type/text object, or a (incorrect) string.
+        Used for Work/Edition 'notes' and 'description' and Author 'bio'.
+        """
+        try:
+            return text.get('value')
+        except:
+            return text
+
+    @staticmethod
+    def get_type(olid):
+        ol_types = {'OL..A': 'author', 'OL..M': 'book', 'OL..W': 'work'}
+        kind = re.sub('\d+', '..', olid)
+        return ol_types[kind]
+
+    @staticmethod
+    def full_key(olid):
+        return "/%ss/%s" % (OpenLibrary.get_type(olid), olid)
```

Comment:
something feel strange about this, especially if it suddenly gets used (or assumed) by other types which grammatically don't fit. Maybe it's fine?